x/ecocredit/marketplace/msg_allowed_denom_proposal.go

```diff
@@ -0,0 +1,42 @@
+package marketplace
+
+import (
+	"fmt"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+var _ govtypes.Content = &AllowedDenomProposal{}
+
+const (
+	AskDenomProposalType = "AllowedDenomProposal"
+)
+
+func init() {
+	govtypes.RegisterProposalType(AskDenomProposalType)
+}
+
+func (m AllowedDenomProposal) ProposalRoute() string { return ecocredit.RouterKey }
+
+func (m AllowedDenomProposal) ProposalType() string { return AskDenomProposalType }
+
+func (m AllowedDenomProposal) ValidateBasic() error {
+	if m.AllowedDenom == nil {
+		return sdkerrors.ErrInvalidRequest.Wrap("allowed_denom cannot be nil")
```

Comment:
Depending on how the message is submitted, it could be `AllowedDenom` or `allowed_denom`. I think it would be safe to remove the underscore, which is how we write errors in other message validation.

```suggestion
		return sdkerrors.ErrInvalidRequest.Wrap("allowed denom cannot be nil")
```